### PR TITLE
Added support for POS embedding

### PR DESCRIPTION
**Pull request by yechenh**

### CODE REVIEW

File: streamline_training.py [hunk 1]
```diff
@@ -16,16 +16,68 @@
...
+        bert_type: the type of Bert. There are two types: Base and POS
+        t2t: If True, will use autoregressive decoder. If False, will use feedforward decoder.
+        number_of_layer: number of hidden layers in the decoder.
+        hidden_size: the size of hidden layer in the decoder.
+        attention_heads: number of attention heads in the decoder. 
+        filter_size: the size of the decoder filter.
+        full_attention: whether to use full attention in the decoder. 
+    """
+    
+    if bert_type == "Base":
```
Comment (XiangZeng2020) [inline streamline_training.py:42]: Define global names for these types at the beginning of the file.
BERT_TYPE_BASE = "Base"
BERT_TYPE_POS = "pos"

File: streamline_training.py [hunk 2]
```diff
@@ -16,16 +16,68 @@
...
+        vocab_type = 2
+        vocab_size = 28996
+    elif bert_type == "POS":
+        vocab_type = 42
```
Comment (XiangZeng2020) [inline streamline_training.py:46]: where are these numbers 2, 42 from?

File: streamline_training.py [hunk 3]
```diff
@@ -16,16 +16,68 @@
...
+import json
+
+from typing import Dict, Text
 
 """Train LaserTagger model and export to GCP bucket."""
 
 GCP_BUCKET = "gs://trained_models_yechen/"
 
 
+def write_lasertagger_config(output_dir: Text, bert_type: Text, t2t: bool, number_of_layer: int, hidden_size: int, attention_heads: int, filter_size: int, full_attention: bool):
```
Comment (XiangZeng2020) [inline streamline_training.py:28]: rename to export_lasertagger_config_to_json

File: streamline_training.py [hunk 4]
```diff
@@ -211,26 +276,23 @@ def __export_to_bucket():
...
-                            absolute path to the folder where the lasertagger
-                            scripts are located
-      abs_path_to_bert      absolute path to the folder where the pretrained BERT
-                            is located
+                            absolute path to the folder where the lasertagger scripts are located
+      abs_path_to_bert      absolute path to the folder where the pretrained BERT is located
       training_file         path to training samples
       tuning_file           path to tuning samples
-    
+      embedding_type        type of embedding.
```
Comment (XiangZeng2020) [inline streamline_training.py:293]: add more details 

type of embedding, must be one of ["POS", ...

File: streamline_training.py [hunk 5]
```diff
@@ -111,20 +163,32 @@ def __preprocess(args):
...
+    if args.embedding_type == "POS":
+        bert_type = "POS"
+    else:
+        bert_type = "Base"
+    
+    write_lasertagger_config(output_dir, bert_type, args.t2t, args.number_layer, args.hidden_size, args.num_attention_head, args.filter_size, args.full_attention)
+    config_file_path = "{}/bert_config.json".format(output_dir)
+    
     print("------ Start training ------")
     f = open(output_dir + "/train.tf_record.num_examples.txt", "r")
```
Comment (XiangZeng2020) [inline streamline_training.py:193]: use with statement to narrow the scope of the variables

with open(output_dir + "/train.tf_record.num_examples.txt", "r") as f:
  num_train_examples = f.read()

File: streamline_training_test.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+import unittest
+from unittest import TestCase
+import streamline_training
+
+from typing import Dict, Text
+
+TEMP_FOLDER_PATH = "~/temp_test_streamline_training"
+    
+    
+class StreamlineTrainingTestCase(TestCase):
```
Comment (XiangZeng2020) [inline streamline_training_test.py:31]: just
StreamlineTrainingTest

File: streamline_training_test.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    def __clean_up(self):
+        subprocess.call(['rm', '-rf', os.path.expanduser(TEMP_FOLDER_PATH)], cwd=os.path.expanduser('~'))
+        
+    def __create_temp_folder(self):
+        subprocess.call(['rm', '-rf', os.path.expanduser(TEMP_FOLDER_PATH)], cwd=os.path.expanduser('~'))
+        subprocess.call(['mkdir', os.path.expanduser(TEMP_FOLDER_PATH)], cwd=os.path.expanduser('~'))
+
+
+    def __generate_correct_output(self, vocab_size: int, vocab_type: int, t2t: bool, number_of_layer: int, hidden_size: int, attention_heads: int, filter_size: int, full_attention: bool) -> Dict:
+        correct_output = {
```
Comment (XiangZeng2020) [inline streamline_training_test.py:41]: expected_output

File: streamline_training.py (hunk 1 above)
Comment (yechenh) [reply streamline_training.py]: Done.

File: streamline_training.py (hunk 2 above)
Comment (yechenh) [reply streamline_training.py]: 2 and 42 are the vocab type for segment id embedding. So for regular Bert, it is just 0 and 1 and thus 2. For the Bert trained with POS, there are 42 different POS tags, and thus the type is 42.

File: streamline_training.py (hunk 3 above)
Comment (yechenh) [reply streamline_training.py]: Done.

File: streamline_training.py (hunk 4 above)
Comment (yechenh) [reply streamline_training.py]: Done.

File: streamline_training.py (hunk 5 above)
Comment (yechenh) [reply streamline_training.py]: Done.

File: streamline_training_test.py (hunk 6 above)
Comment (yechenh) [reply streamline_training_test.py]: Done.

File: streamline_training_test.py (hunk 7 above)
Comment (yechenh) [reply streamline_training_test.py]: Done.

File: streamline_training.py (hunk 2 above)
Comment (XiangZeng2020) [reply streamline_training.py]: Got it, you should add comment to explain these numbers or the reader will be confused.
Also it is better to give a pointer to the source of these tags.

**Pull request merged**